.github/workflows/pr-smoke-test.yml

```diff
@@ -0,0 +1,59 @@
+name: Run smoke tests against pull request.
+on:
+  pull_request:
+    types:
+      - labeled
+jobs:
+  prcheck:
+    name: Smoke test a pull request.
+    if: "${{ contains(github.event.label.name, 'run: smoke tests') }}"
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Create dirs.
+        run: |
+              mkdir -p code/woocommerce
+              mkdir -p package/woocommerce
+              mkdir -p tmp/woocommerce
+              mkdir -p node_modules
+
+      - name: Checkout code.
+        uses: actions/checkout@v2
+        with:
+          ref: trunk
+
+      - name: Install prerequisites.
+        run: |
+          npm install
+          composer install --no-dev
+          npm run build:assets
+          npm install jest
+          
+      - name: Run smoke test.
+        env:
+          SMOKE_TEST_URL: ${{ secrets.SMOKE_TEST_URL }}
+          SMOKE_TEST_ADMIN_USER: ${{ secrets.SMOKE_TEST_ADMIN_USER }}
+          SMOKE_TEST_ADMIN_PASSWORD: ${{ secrets.SMOKE_TEST_ADMIN_PASSWORD }}
+          SMOKE_TEST_ADMIN_USER_EMAIL: ${{ secrets.SMOKE_TEST_ADMIN_USER_EMAIL }}
+          SMOKE_TEST_CUSTOMER_USER: ${{ secrets.SMOKE_TEST_CUSTOMER_USER }}
+          SMOKE_TEST_CUSTOMER_PASSWORD: ${{ secrets.SMOKE_TEST_CUSTOMER_PASSWORD }}
+          WC_E2E_SCREENSHOTS: 1
+          E2E_RETEST: 1
+          E2E_SLACK_TOKEN: ${{ secrets.SMOKE_TEST_SLACK_TOKEN }}
+          E2E_SLACK_CHANNEL: ${{ secrets.SMOKE_TEST_SLACK_CHANNEL }}
+          UPDATE_WC: 1
+          DEFAULT_TIMEOUT_OVERRIDE: 120000
+        run: |
+          npx wc-e2e test:e2e ./tests/e2e/specs/smoke-tests/update-woocommerce.js
+          npx wc-e2e test:e2e
+
+      - name: Remove label from pull request.
+        if: always()
```

Comment:
@zhongruige This fatals and indicates the tests failed if the label is not present. https://github.com/woocommerce/woocommerce/blob/trunk/.github/workflows/update-feedback-labels.yml#L10 has the condition for checking for a label.

Reply:
Thanks! The label was present during the run, so I am curious on why it returned a 404, which I'll need to dig into a little bit as well.

Reply:
Looks like the workflow is removing the label now 🎉 Thanks for testing that @rrennick.

Looking into the way the [status checks work in workflows]( https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions), the success status check is always applied:

> A default status check of success() is applied unless you include one of these functions

So, it seems like we can update it to be the label check, e.g. `if always() && "${{ contains(github.event.label.name, 'run: smoke tests') }}"`. Let me know your thoughts.

Reply:
> Let me know your thoughts.

Sounds like it will work.

Reply:
Sounds good--I've pushed up that change: https://github.com/woocommerce/woocommerce/pull/30698/files#diff-601f4659fb75d85e41f0c206e3d7c72075a0f47851efd40b47b2fb38e74a8571R51

Reply:
Unfortunately it didn't end up working and it skipped the step:

<img width="409" alt="Screen Shot 2021-09-23 at 09 24 02" src="https://user-images.githubusercontent.com/71906536/134536556-08f5d1d3-e64d-4359-8ced-fbe7af430ffd.png">

So, I've switched it back to the previous `if: always()` check that was there.

Reply:
@zhongruige I created #30807 to test changes to this PR without adding any more commits to it. I found three things

- `github.issues.removeLabel(` was throwing an api error even though it successfully removed the label. My PR uses the same action as `Update contributor feedback labels on comment` to remove the label
- This check was testing against the label that triggered the event vs checking the labels on the PR:
```
always() && 
          "${{ contains(github.event.label.name, 'run: smoke tests') }}"
```
- The action is running against a PR vs an issue and I'm guessing the event doesn't have an `issue.labels` array. That would cause this test to fail or error:
```
always() && 
          "${{ contains(github.event.issue.labels.*.name, 'run: smoke tests') }}"
```

Once you have confirmed the logic on my PR works correctly, you can add it to this PR and delete mine.

Reply:
Nice I like that, and the usage of `actions-ecosystem/action-remove-labels@v1` instead. This makes it a lot cleaner. I brought in those changes and ready for review.